linera-execution/src/execution_state_actor.rs

```diff
@@ -358,6 +358,11 @@ where
                 }
             }
 
+            NextApplicationIndex { callback } => {
+                let index = self.txn_tracker.application_index();
```

Comment:
~Let's call this `next_application_index`, too.~

Ah, I see! `next_application_index` already exists, but that one increments the index!
Then the naming we currently have is inconsistent.

We could take inspiration from `Iterator`, and call them `next_application_index` (mutates) and `peek_application_index` (does not mutate). Either way, let's make the actor request match the function name everywhere.